dolphinscheduler-registry/dolphinscheduler-registry-plugins/dolphinscheduler-registry-raft/src/main/java/org/apache/dolphinscheduler/plugin/registry/raft/client/RaftRegistryClient.java

```diff
@@ -0,0 +1,221 @@
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import lombok.extern.slf4j.Slf4j;
+
+import com.alipay.sofa.jraft.rhea.client.DefaultRheaKVStore;
+import com.alipay.sofa.jraft.rhea.client.RheaKVStore;
+import com.alipay.sofa.jraft.rhea.options.PlacementDriverOptions;
+import com.alipay.sofa.jraft.rhea.options.RegionRouteTableOptions;
+import com.alipay.sofa.jraft.rhea.options.RheaKVStoreOptions;
+import com.alipay.sofa.jraft.rhea.options.configured.MultiRegionRouteTableOptionsConfigured;
+import com.alipay.sofa.jraft.rhea.options.configured.PlacementDriverOptionsConfigured;
+import com.alipay.sofa.jraft.rhea.options.configured.RheaKVStoreOptionsConfigured;
+import com.alipay.sofa.jraft.rhea.storage.KVEntry;
+
+@Slf4j
+public class RaftRegistryClient implements IRaftRegistryClient {
+
+    private final RheaKVStore rheaKvStore;
+    private final RaftRegistryProperties raftRegistryProperties;
+    private final IRaftConnectionStateManager raftConnectionStateManager;
+    private final IRaftSubscribeDataManager raftSubscribeDataManager;
+    private final IRaftLockManager raftLockManager;
+    private volatile boolean started;
+    private static final String MASTER_MODULE = "master";
+    public RaftRegistryClient(RaftRegistryProperties raftRegistryProperties) {
+        this.raftRegistryProperties = raftRegistryProperties;
+        this.rheaKvStore = new DefaultRheaKVStore();
+        this.raftConnectionStateManager = new RaftConnectionStateManager(raftRegistryProperties);
+        this.raftSubscribeDataManager = new RaftSubscribeDataManager(raftRegistryProperties, rheaKvStore);
+        this.raftLockManager = new RaftLockManager(rheaKvStore, raftRegistryProperties);
+
+        initRheakv();
+    }
+
+    private void initRheakv() {
+        final List<RegionRouteTableOptions> regionRouteTableOptionsList = MultiRegionRouteTableOptionsConfigured
+                .newConfigured()
+                .withInitialServerList(-1L /* default id */, raftRegistryProperties.getServerAddressList())
+                .config();
+        final PlacementDriverOptions pdOpts = PlacementDriverOptionsConfigured.newConfigured()
+                .withFake(true)
+                .withRegionRouteTableOptionsList(regionRouteTableOptionsList)
+                .config();
+        final RheaKVStoreOptions opts = RheaKVStoreOptionsConfigured.newConfigured() //
+                .withClusterName(raftRegistryProperties.getClusterName()) //
+                .withPlacementDriverOptions(pdOpts) //
+                .config();
+        this.rheaKvStore.init(opts);
+    }
+
+    @Override
+    public void start() {
+        if (this.started) {
+            log.info("RaftRegistryClient is already started");
+            return;
+        }
+        log.info("starting raft client registry...");
+        if (raftRegistryProperties.getModule().equals(MASTER_MODULE)) {
+            raftSubscribeDataManager.start();
+        }
```

Comment:
Why only master start the `raftSubscribeDataManager`, this is strange.

Reply:
done